packages/babel-compat-data/scripts/utils-build-data.js

```diff
@@ -149,15 +148,19 @@ exports.generateData = (environments, features) => {
 };
 
 exports.writeFile = function (data, dataPath, name) {
+  dataPath = process.env.BABEL_8_BREAKING
+    ? dataPath.replace(".json", "-next.json")
+    : dataPath;
   const stringified = JSON.stringify(data, null, 2) + "\n";
   if (process.env.CHECK_COMPAT_DATA) {
     const currentData = fs.readFileSync(dataPath, "utf8");
 
     // Compare as JSON strings to also check keys ordering
     if (currentData !== stringified) {
```

Comment:
Since the Babel 7 data is a superset of the Babel 8 data (thus the Babel 8 build would also work with the Babel 7 data), can we instead only generate the Babel 8 file when `process.env.BABEL_8_BREAKING && process.env.IS_PUBLISH`, and skip this check in that case?